app/javascript/onboarding/components/FollowUsers.jsx

```diff
@@ -176,40 +176,45 @@ class FollowUsers extends Component {
             </header>
 
             <div data-testid="onboarding-users">
-              {users.map((user) => (
-                <button
-                  data-testid="onboarding-user-button"
-                  type="button"
-                  onClick={() => this.handleClick(user)}
-                  onKeyDown={() => this.handleKeyDown(user)}
-                  className={
-                    selectedUsers.includes(user)
-                      ? 'user content-row selected'
-                      : 'user content-row unselected'
-                  }
-                >
-                  <figure className="user-avatar-container">
-                    <img
-                      className="user-avatar"
-                      src={user.profile_image_url}
-                      alt="profile"
-                    />
-                  </figure>
-                  <div className="user-info">
-                    <h4 className="user-name">{user.name}</h4>
-                    <p className="user-summary">
-                      {he.unescape(user.summary || '')}
-                    </p>
-                  </div>
+              {users.map((user) => {
```

Comment:
Refactor (blocking): use checkboxes instead

> 3. The same tag follow buttons don't have any semantic information marking them as selected, and they are nested in DIV elements so screen reader users won't know how many tag buttons there are, nor when they are or aren't selected.

@Rafi993, I think instead of  keeping these as buttons, we can change them to checkboxes so that they they have more semantic meaning (knowing that they are selected or not) and we can give them labels so we don't even need to use the `aria-*` attributes.

Since the checkbox will have an associated label, we can visually hide the actual checkbox and clicking the label will still check/uncheck it.

As well the `<div data-testid="onboarding-users">` can become a `<fieldset />` and one of the headings above can most likely become the legend for the `<fieldset />`.

If you need some assistance making this work with checkboxes let us know.

Not for this PR, but when you tackle the other part of the issue, the same thing can be done for the tags to follow.

Reply:
@nickytonline did you styling label for checkbox like the existing follow button and hiding the checkbox?

Reply:
something like this 

```html
                    <label htmlFor={user.name} className="relative">
                      <input
                        type="checkbox"
                        checked={selected}
                        className="user-following-status invisible absolute top-0 bottom-0 right-0 left-0"
                        id={user.name}
                      />
                      <div
                        tabIndex="0"
                        data-testid="onboarding-user-following-status"
                        aria-label={`Follow ${user.name}`}
                        aria-checked={selected}
                        role="checkbox"
                      >
                        {selected ? 'Following' : 'Follow'}
                      </div>
                    </label>
 ```
                   
and corresponding css

```css

    input[type='checkbox']:not(:checked).user-following-status + div {
      @extend .crayons-btn;
      @extend .crayons-btn--outlined;
    }

    input[type='checkbox']:checked.user-following-status + div {
      @extend .crayons-btn;
    }

```

which now renders like this

![checkbox](https://user-images.githubusercontent.com/8092120/106150737-a58c5b00-61a1-11eb-87e8-71259483984d.png)

Reply:
Hi @Rafi993 using checkboxes in this way is giving some better semantics to the UI, however it's still not being picked up by screen reader because the whole row for each suggested person is wrapped in a button (line 183).

The effect is that the screen reader will only announce the button on line 183, and not the checkbox inside of it (interactive items within interactive items like this is invalid HTML for this reason).

I think what you have on the checkboxes so far is looking good, but this needs some further refactoring so we no longer have that wrapping button, and the relevant row content forms part of the checkbox label. Let me know if you want to discuss further or need some assistance 👍

Reply:
I have removed the outer button now